### PR TITLE
BLOCKCHAIN-443 FE senate

### PR DESCRIPTION
**Pull request by SzymonZiebaNeti**
Wait until `https://github.com/liberland/liberland_substrate/pull/384` is merged

### CODE REVIEW

File: src/api/nodeRpcCall.js [hunk 1]
```diff
@@ -2034,19 +2077,19 @@ const getScheduledCalls = async () => {
...
         ...item,
         call: item.call.unwrap(),
-      })))
-    .filter((item) => item.call.call.isLookup // our FE only does lookups now
-      && item.call.maybePeriodic.isNone); // we're interested only in referendum results, so nonperiodic
+      })));
 
+  const lookupItems = agendaItems.filter((item) => item.call.call.isLookup // our FE only does lookups now
+      && item.call.maybePeriodic.isNone); // we're interested only in referendum results, so nonperiodic
   // we only want do download small preimages. fetching multi-megabyte setCode could be painful.
```
Comment (DorianSternVukotic) [inline src/api/nodeRpcCall.js:2084]: Maybe we should make a note and display to the user if there are lookupitems that are bigger, but just with a link to polkadotjs where he can see it. Is it possible to get something around the users if a call is purposefully big ?

File: src/api/nodeRpcCall.js [hunk 2]
```diff
@@ -2420,6 +2485,46 @@ const getStakingData = async (walletAddress) => {
   return { stakingInfo, sessionProgress };
 };
 
+const getSenateMembers = async () => {
```
Comment (DorianSternVukotic) [inline src/api/nodeRpcCall.js:2488]: are senate members displayed anywhere? I couldnt find it

File: src/components/Congress/Motions/index.js [hunk 3]
```diff
@@ -22,132 +15,22 @@ export default function Motions() {
     dispatch(congressActions.getMotions.call());
   }, [dispatch]);
 
+  if (!motions || motions.length < 1) {
+    return (<div>There is no any motion</div>);
```
Comment (DorianSternVukotic) [inline src/components/Congress/Motions/index.js:19]: Rename to There are no open motions

File: src/components/Modals/SpendModal.js [hunk 4]
```diff
@@ -36,11 +43,10 @@ function SpendAssetModal({ closeModal, assetData }) {
...
 
   const transfer = (values) => {
-    dispatch(congressActions.congressSendAssets.call({
+    dispatch(onSend({
       transferToAddress: values.recipient,
-      transferAmount: parseAssets(values.amount, assetData.metadata.decimals),
-      assetData,
-      remarkInfo: values.description || `Congress spend ${assetData.metadata.name} (${assetData.metadata.symbol})`,
+      transferAmount: parseMerits(values.amount),
+      remarkInfo: values.description || `${isCongress ? 'Congress' : 'Senate'} ${title}`,
```
Comment (DorianSternVukotic) [inline src/components/Modals/SpendModal.js:49]: Interface always shows congress, even when in senate tab

File: src/components/Modals/SpendModal.js [hunk 5]
```diff
@@ -103,8 +106,18 @@ function SpendAssetModal({ closeModal, assetData }) {
...
-      <div className={styles.description}>How long will it take Congress to close the motion?</div>
+      <div className={styles.title}>
+        {isCongress ? 'Congress' : 'Senate'}
+        {' '}
+        voting time in days
+      </div>
+      <div className={styles.description}>
+        How long will it take
+        {' '}
+        {isCongress ? 'Congress' : 'Senate'}
```
Comment (DorianSternVukotic) [inline src/components/Modals/SpendModal.js:117]: interface always shows congress, even in senate tab

File: src/components/Senate/Motions/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+function Motions() {
+  const dispatch = useDispatch();
+  const motions = useSelector(senateSelectors.motions);
+  useEffect(() => {
+    dispatch(senateActions.senateGetMotions.call());
+  }, [dispatch]);
+
+  if (!motions || motions.length < 1) {
+    return (<div>There is no any motion</div>);
```
Comment (DorianSternVukotic) [inline src/components/Senate/Motions/index.js:15]: rename to There are no open motions

File: src/components/Senate/ScheduledCongressSpending/index.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  const scheduledCalls = useSelector(senateSelectors.scheduledCalls);
+
+  useEffect(() => {
+    dispatch(senateActions.senateGetCongressSpending.call());
+  }, [dispatch]);
+
+  const { preimagesInside, lookupItemsData } = scheduledCalls;
+
+  if (!scheduledCalls || (preimagesInside.length < 1 && lookupItemsData.length < 1)) {
+    return (<div>There is no any item</div>);
```
Comment (DorianSternVukotic) [inline src/components/Senate/ScheduledCongressSpending/index.js:23]: Rename to there are no open items

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]

**Comment by DorianSternVukotic** [comment]
Also fix

- sending asset as senate doesnt work, wrong amount error
- sending asest interface asks for amount LLD
- add remark to sending asset
- test congress functionalities if something broke

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

File: src/components/Modals/SendAssetModal.js [hunk 8]
```diff
@@ -5,46 +5,75 @@ import { useForm } from 'react-hook-form';
...
   return (
     <form className={styles.getCitizenshipModal} onSubmit={handleSubmit(transfer)}>
       <div className={styles.h3}>
         Send
+        {' '}
         {assetData.metadata.symbol}
       </div>
       <div className={styles.description}>
-        You are going to send tokens from your wallet
+        {!isRemarkNeeded
```
Comment (DorianSternVukotic) [inline src/components/Modals/SendAssetModal.js:74]: OOF

We ll need to refactor how modal decides whether its sending your tokens or govt tokens

**Pull request merged**